File: backend/src/main/java/com/ll/nbe342team8/domain/book/review/dto/ReviewResponseDto.java
package com.ll.nbe342team8.domain.book.review.dto;


import com.ll.nbe342team8.domain.book.review.entity.Review;

import java.time.LocalDateTime;

public record ReviewResponseDto(
        Long bookId,
        Long reviewId,
        Long memberId,
        String content,
        Double rating,
        LocalDateTime createDate,
        LocalDateTime modifyDate
) {

    public static ReviewResponseDto from(Review review){
        return new ReviewResponseDto(
                review.getBook().getId(),
                review.getId(),
                review.getMember().getId(),
                review.getContent(),
                review.getRating(),
                review.getCreateDate(),
                review.getModifyDate()
        );
    }
}

File: backend/src/main/java/com/ll/nbe342team8/domain/book/review/dto/ReviewResponseDto.kt
package com.ll.nbe342team8.domain.book.review.dto

import com.ll.nbe342team8.domain.book.review.entity.Review
import java.time.LocalDateTime

@JvmRecord
data class ReviewResponseDto(
    val bookId: Long?,
    val reviewId: Long?,
    val memberId: Long?,
    val content: String?,
    val rating: Double?,
    val createDate: LocalDateTime?,
    val modifyDate: LocalDateTime?
) {
    companion object {
        @JvmStatic
        fun from(review: Review): ReviewResponseDto {
            return ReviewResponseDto(
                review.book.id,
                review.id,
                review.member.id,
                review.content,
                review.rating,
                review.createDate,
                review.modifyDate
            )
        }
    }
}
